### ISSUE TITLE
Path Of Exile crash with Proton 4.11

### ISSUE THREAD

**Issue by phush0** [daniel-schuermann/mesa#93]
**Describe the bug**
Game crashes after selecting character

**Screenshots**
For rendering errors, please insert screenshots showing incorrect rendering (ACO) and correct rendering (LLVM).

**System information:**
 - GPU: VegaM (8706G)
 - ACO version: 19.1.0+aco+git1564433637-5c6077221dd-1bionic2

**Additional context**
```
fExile_x64Steam.exe: ../src/amd/compiler/aco_insert_exec_mask.cpp:984: void aco::{anonymous}::process_block(aco::{anonymous}::exec_ctx&, aco::Block*): Assertion `block->index != ctx.program->blocks.size() - 1 || ctx.info[block->index].exec.size() <= 2' failed.
wine: Assertion failed at address 0x7fecab6dbe97 (thread 0045), starting debugger...
ERROR: ld.so: object '/home/phusho/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
>>> Adding process 18592 for game ID 238960
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
Unhandled exception: assertion failed in 64-bit code (0x00007fecab6dbe97).
0056:fixme:dbghelp:elf_search_auxv can't find symbol in module
Register dump:
 rip:00007fecab6dbe97 rsp:000000000a44bf20 rbp:00007fecab8547d8 eflags:00000246 (   - --  I  Z- -P- )
 rax:0000000000000000 rbx:0000000000000000 rcx:00007fecab6dbe97 rdx:0000000000000000
 rsi:000000000a44bf20 rdi:0000000000000002  r8:0000000000000000  r9:000000000a44bf20 r10:0000000000000008
 r11:0000000000000246 r12:00007fec6a910a60 r13:00007fec6a911030 r14:00000000000003d8 r15:00000000000001cb
Stack dump:
0x000000000a44bf20:  0000000000000000 00007fec49abae30
0x000000000a44bf30:  00000000000001cb 00007fecab7255e0
0x000000000a44bf40:  00007fecfbad8000 00007fec49abae30
0x000000000a44bf50:  00007fec49abae95 00007fec49abae30
0x000000000a44bf60:  00007fec49abae30 00007fec49abaf35
0x000000000a44bf70:  00007fec49abaf5c 00007fec49abae30
0x000000000a44bf80:  00007fec49abaf5c 0000000000000000
0x000000000a44bf90:  0000000000000000 0000000000000000
0x000000000a44bfa0:  fffffffe7fffffff ffffffffffffffff
0x000000000a44bfb0:  ffffffffffffffff ffffffffffffffff
0x000000000a44bfc0:  ffffffffffffffff ffffffffffffffff
0x000000000a44bfd0:  ffffffffffffffff ffffffffffffffff
Backtrace:
=>0 0x00007fecab6dbe97 execlp+0xffffffffffffffff() in libc.so.6 (0x00007fecab8547d8)
0x00007fecab6dbe97 execlp+0xffffffffffffffff in libc.so.6: movq	0x0000000000000108(%rsp),%rcx
Modules:
Module	Address					Debug info	Name (186 modules)
PE	          930000-          b97000	Deferred        d3d11
PE	          ef0000-          ef3000	Deferred        api-ms-win-core-fibers-l1-1-1
PE	          f50000-          f53000	Deferred        api-ms-win-core-localization-l1-2-1
PE	         4ad0000-         4b33000	Deferred        bink2w64
PE	         5580000-         56d5000	Deferred        fmodstudio64
PE	         5970000-         5a06000	Deferred        openvr_api_dxvk
PE	         a460000-         a864000	Deferred        d3dcomp_47_x64
PE	        3b400000-        3b421000	Deferred        steam_api64
PE	        64940000-        64943000	Deferred        shcore
PE	        65000000-        65028000	Deferred        ole32
PE	        65680000-        65683000	Deferred        api-ms-win-core-sysinfo-l1-2-1
PE	        66440000-        66449000	Deferred        msacm32
PE	        66500000-        66504000	Deferred        rsaenh
PE	        66780000-        66783000	Deferred        api-ms-win-core-string-l1-1-0
PE	        68a40000-        68a48000	Deferred        shlwapi
PE	        69c80000-        69c84000	Deferred        propsys
PE	        6a100000-        6a108000	Deferred        oleaut32
PE	        6a700000-        6a70f000	Deferred        setupapi
PE	        6b6c0000-        6b76c000	Deferred        comdlg32
PE	        6bec0000-        6bec4000	Deferred        imm32
PE	        6d3c0000-        6d3c4000	Deferred        dsound
PE	        6dc00000-        6dc03000	Deferred        api-ms-win-appmodel-runtime-l1-1-1
PE	        6e340000-        6e343000	Deferred        api-ms-win-core-synch-l1-2-0
PE	        6e6c0000-        6e713000	Deferred        comctl32
PE	        6f200000-        6f397000	Deferred        dxgi
PE	        6fbc0000-        6fbc4000	Deferred        rpcrt4
PE	        70140000-        70144000	Deferred        mmdevapi
PE	        71040000-        71043000	Deferred        kernelbase
ELF	        7b400000-        7b86c000	Deferred        kernel32<elf>
  \-PE	        7b420000-        7b86c000	\               kernel32
ELF	        7bc00000-        7bd80000	Deferred        ntdll<elf>
  \-PE	        7bc20000-        7bd80000	\               ntdll
ELF	        7c000000-        7c003000	Deferred        <wine-loader>
PE	       140000000-       142004000	Deferred        pathofexile_x64steam
PE	       180000000-       180206000	Deferred        fmod64
ELF	    7fec603b3000-    7fec64000000	Deferred        libllvm-8.so.1
ELF	    7fec68287000-    7fec68393000	Deferred        windowscodecs<elf>
  \-PE	    7fec682b0000-    7fec68393000	\               windowscodecs
ELF	    7fec69899000-    7fec69ade000	Deferred        libvklayer_steam_fossilize.so
ELF	    7fec69ade000-    7fec69cf4000	Deferred        steamoverlayvulkanlayer.so
ELF	    7fec69cf4000-    7fec69f1e000	Deferred        libtinfo.so.5
ELF	    7fec69f1e000-    7fec6a155000	Deferred        libedit.so.2
ELF	    7fec6a155000-    7fec6a36f000	Deferred        libelf.so.1
ELF	    7fec6a36f000-    7fec6a579000	Deferred        libdrm_amdgpu.so.1
ELF	    7fec6a579000-    7fec6adbe000	Deferred        libvulkan_radeon.so
ELF	    7fec6adbe000-    7fec6afc0000	Deferred        libxshmfence.so.1
ELF	    7fec6afc0000-    7fec6b1c7000	Deferred        libxcb-sync.so.1
ELF	    7fec6b1c7000-    7fec6b3ca000	Deferred        libxcb-present.so.0
ELF	    7fec6b3ca000-    7fec6b5cc000	Deferred        libx11-xcb.so.1
ELF	    7fec6b5cc000-    7fec6b7dc000	Deferred        libxcb-randr.so.0
ELF	    7fec6b7dc000-    7fec6c000000	Deferred        libvulkan_intel.so
ELF	    7fec980f9000-    7fec98308000	Deferred        libwayland-client.so.0
ELF	    7fec98308000-    7fec9850c000	Deferred        libxcb-dri3.so.0
ELF	    7fec9850c000-    7fec9871d000	Deferred        libdrm.so.2
ELF	    7fec9871d000-    7fec98945000	Deferred        libpng12.so.0
ELF	    7fec9894c000-    7fec98bf5000	Deferred        libvorbisenc.so.2
ELF	    7fec98bf5000-    7fec98e20000	Deferred        libvorbis.so.0
ELF	    7fec98e20000-    7fec99029000	Deferred        libogg.so.0
ELF	    7fec99029000-    7fec992a0000	Deferred        libflac.so.8
ELF	    7fec992a0000-    7fec994a6000	Deferred        libasyncns.so.0
ELF	    7fec994a6000-    7fec9971f000	Deferred        libsndfile.so.1
ELF	    7fec9971f000-    7fec99929000	Deferred        libwrap.so.0
ELF	    7fec99929000-    7fec99ba8000	Deferred        libpulsecommon-12.2.so
ELF	    7fec99ba8000-    7fec99df8000	Deferred        libpulse.so.0
ELF	    7fec99df8000-    7fec99e2c000	Deferred        winepulse<elf>
  \-PE	    7fec99e00000-    7fec99e2c000	\               winepulse
ELF	    7fec99e2c000-    7fec9a033000	Deferred        libnss_dns.so.2
ELF	    7fec9a033000-    7fec9a236000	Deferred        libnss_mdns4_minimal.so.2
ELF	    7fec9b81c000-    7fec9d42d000	Deferred        steamclient.so
ELF	    7fec9d42d000-    7fec9d75f000	Deferred        lsteamclient<elf>
  \-PE	    7fec9d530000-    7fec9d75f000	\               lsteamclient
ELF	    7fec9d75f000-    7fec9d7b8000	Deferred        wbemprox<elf>
  \-PE	    7fec9d770000-    7fec9d7b8000	\               wbemprox
ELF	    7fec9d7b8000-    7fec9d9cd000	Deferred        libgpg-error.so.0
ELF	    7fec9d9cd000-    7fec9dce8000	Deferred        libgcrypt.so.20
ELF	    7fec9dce8000-    7fec9df04000	Deferred        liblz4.so.1
ELF	    7fec9df04000-    7fec9e12a000	Deferred        liblzma.so.5
ELF	    7fec9e12a000-    7fec9e3ae000	Deferred        libsystemd.so.0
ELF	    7fec9e3ae000-    7fec9e5b2000	Deferred        libkeyutils.so.1
ELF	    7fec9e5b2000-    7fec9e7ff000	Deferred        libdbus-1.so.3
ELF	    7fec9e7ff000-    7fec9ea0a000	Deferred        libkrb5support.so.0
ELF	    7fec9ea0a000-    7fec9ec3c000	Deferred        libk5crypto.so.3
ELF	    7fec9ec3c000-    7fec9ef12000	Deferred        libkrb5.so.3
ELF	    7fec9ef12000-    7fec9f123000	Deferred        libavahi-client.so.3
ELF	    7fec9f123000-    7fec9f32f000	Deferred        libavahi-common.so.3
ELF	    7fec9f32f000-    7fec9f57a000	Deferred        libgssapi_krb5.so.2
ELF	    7fec9f57a000-    7fec9f806000	Deferred        libcups.so.2
ELF	    7fec9f806000-    7fec9fa5a000	Deferred        libvulkan.so.1
ELF	    7fec9fa5a000-    7fec9fa9d000	Deferred        uxtheme<elf>
  \-PE	    7fec9fa60000-    7fec9fa9d000	\               uxtheme
ELF	    7fec9fa9f000-    7fec9fca5000	Deferred        libxfixes.so.3
ELF	    7fec9fca5000-    7fec9feaf000	Deferred        libxcursor.so.1
ELF	    7fec9feaf000-    7feca00bf000	Deferred        libxi.so.6
ELF	    7feca00bf000-    7feca02c2000	Deferred        libxcomposite.so.1
ELF	    7feca02c2000-    7feca04cd000	Deferred        libxrandr.so.2
ELF	    7feca04cd000-    7feca06d7000	Deferred        libxrender.so.1
ELF	    7feca06d7000-    7feca08dd000	Deferred        libxxf86vm.so.1
ELF	    7feca08dd000-    7feca0ae0000	Deferred        libxinerama.so.1
ELF	    7feca0ae0000-    7feca0cf2000	Deferred        libxext.so.6
ELF	    7feca0cf2000-    7feca0db9000	Deferred        winex11<elf>
  \-PE	    7feca0d00000-    7feca0db9000	\               winex11
ELF	    7feca0e93000-    7feca10c5000	Deferred        libexpat.so.1
ELF	    7feca10c5000-    7feca130a000	Deferred        libfontconfig.so.1
ELF	    7feca130a000-    7feca153c000	Deferred        libpng16.so.16
ELF	    7feca153c000-    7feca17f0000	Deferred        libfreetype.so.6
ELF	    7feca17f0000-    7feca184e000	Deferred        winspool<elf>
  \-PE	    7feca1800000-    7feca184e000	\               winspool
ELF	    7feca184e000-    7feca18a1000	Deferred        winevulkan<elf>
  \-PE	    7feca1860000-    7feca18a1000	\               winevulkan
ELF	    7feca18a1000-    7feca18ba000	Deferred        vulkan-1<elf>
  \-PE	    7feca18b0000-    7feca18ba000	\               vulkan-1
ELF	    7feca18ba000-    7feca191a000	Deferred        usp10<elf>
  \-PE	    7feca18c0000-    7feca191a000	\               usp10
ELF	    7feca191a000-    7feca194c000	Deferred        iphlpapi<elf>
  \-PE	    7feca1920000-    7feca194c000	\               iphlpapi
ELF	    7feca194c000-    7feca1971000	Deferred        bcrypt<elf>
  \-PE	    7feca1950000-    7feca1971000	\               bcrypt
ELF	    7feca1971000-    7feca1a7a000	Deferred        crypt32<elf>
  \-PE	    7feca1980000-    7feca1a7a000	\               crypt32
ELF	    7feca1a7a000-    7feca1b5a000	Deferred        winmm<elf>
  \-PE	    7feca1a80000-    7feca1b5a000	\               winmm
ELF	    7feca1b5a000-    7feca1bd9000	Deferred        dbghelp<elf>
  \-PE	    7feca1b60000-    7feca1bd9000	\               dbghelp
ELF	    7feca1bd9000-    7feca1de1000	Deferred        libffi.so.6
ELF	    7feca1de1000-    7feca2019000	Deferred        libcrypt.so.1
ELF	    7feca2019000-    7feca2322000	Deferred        libsqlite3.so.0
ELF	    7feca2322000-    7feca256c000	Deferred        libhx509.so.5
ELF	    7feca256c000-    7feca277b000	Deferred        libheimbase.so.1
ELF	    7feca277b000-    7feca29a4000	Deferred        libwind.so.0
ELF	    7feca29a4000-    7feca2c25000	Deferred        libgmp.so.10
ELF	    7feca2c25000-    7feca2e59000	Deferred        libhogweed.so.4
ELF	    7feca2e59000-    7feca308f000	Deferred        libnettle.so.6
ELF	    7feca308f000-    7feca32a2000	Deferred        libtasn1.so.6
ELF	    7feca32a2000-    7feca3620000	Deferred        libunistring.so.2
ELF	    7feca3620000-    7feca383d000	Deferred        libidn2.so.0
ELF	    7feca383d000-    7feca3b6c000	Deferred        libp11-kit.so.0
ELF	    7feca3b6c000-    7feca3d89000	Deferred        libz.so.1
ELF	    7feca3d89000-    7feca3f9f000	Deferred        libroken.so.18
ELF	    7feca3f9f000-    7feca41d5000	Deferred        libhcrypto.so.4
ELF	    7feca41d5000-    7feca43d9000	Deferred        libcom_err.so.2
ELF	    7feca43d9000-    7feca467b000	Deferred        libasn1.so.8
ELF	    7feca467b000-    7feca4908000	Deferred        libkrb5.so.26
ELF	    7feca4908000-    7feca4b11000	Deferred        libheimntlm.so.0
ELF	    7feca4b11000-    7feca4e76000	Deferred        libgnutls.so.30
ELF	    7feca4e76000-    7feca50b7000	Deferred        libgssapi.so.3
ELF	    7feca50b7000-    7feca52d2000	Deferred        libsasl2.so.2
ELF	    7feca52d2000-    7feca54ed000	Deferred        libresolv.so.2
ELF	    7feca54ed000-    7feca56fb000	Deferred        liblber-2.4.so.2
ELF	    7feca56fb000-    7feca594d000	Deferred        libldap_r-2.4.so.2
ELF	    7feca594d000-    7feca59bf000	Deferred        wldap32<elf>
  \-PE	    7feca5960000-    7feca59bf000	\               wldap32
ELF	    7feca59bf000-    7feca5a00000	Deferred        ws2_32<elf>
  \-PE	    7feca59d0000-    7feca5a00000	\               ws2_32
ELF	    7feca5a00000-    7feca5a17000	Deferred        version<elf>
  \-PE	    7feca5a10000-    7feca5a17000	\               version
ELF	    7feca5a17000-    7feca5c9f000	Deferred        user32<elf>
  \-PE	    7feca5a40000-    7feca5c9f000	\               user32
ELF	    7feca5c9f000-    7feca5d90000	Deferred        msvcrt<elf>
  \-PE	    7feca5cc0000-    7feca5d90000	\               msvcrt
ELF	    7feca5d90000-    7feca5f40000	Deferred        gdi32<elf>
  \-PE	    7feca5db0000-    7feca5f40000	\               gdi32
ELF	    7feca5f40000-    7feca698c000	Deferred        shell32<elf>
  \-PE	    7feca5f60000-    7feca698c000	\               shell32
ELF	    7feca6a8c000-    7feca6b3b000	Deferred        advapi32<elf>
  \-PE	    7feca6aa0000-    7feca6b3b000	\               advapi32
ELF	    7feca773b000-    7feca794d000	Deferred        libnss_files.so.2
ELF	    7feca794d000-    7feca7b67000	Deferred        libnsl.so.1
ELF	    7feca7b67000-    7feca7d73000	Deferred        libnss_nis.so.2
ELF	    7feca7d73000-    7feca7f7d000	Deferred        libnss_compat.so.2
ELF	    7feca965c000-    7feca9871000	Deferred        libbsd.so.0
ELF	    7feca9871000-    7feca9a77000	Deferred        libxdmcp.so.6
ELF	    7feca9a77000-    7feca9c7b000	Deferred        libxau.so.6
ELF	    7feca9c7b000-    7feca9ea3000	Deferred        libxcb.so.1
ELF	    7feca9ea3000-    7fecaa1db000	Deferred        libx11.so.6
ELF	    7fecaa1dd000-    7fecaa3f5000	Deferred        libgcc_s.so.1
ELF	    7fecaa3f5000-    7fecaa6ab000	Deferred        libgldispatch.so.0
ELF	    7fecaa6ab000-    7fecaa8dc000	Deferred        libglx.so.0
ELF	    7fecaa8dc000-    7fecaac7a000	Deferred        libm.so.6
ELF	    7fecab005000-    7fecab209000	Deferred        libdl.so.2
ELF	    7fecab209000-    7fecab411000	Deferred        librt.so.1
ELF	    7fecab411000-    7fecab69d000	Deferred        libgl.so.1
ELF	    7fecab69d000-    7fecaba8e000	Export          libc.so.6
ELF	    7fecaba8e000-    7fecabcad000	Deferred        libpthread.so.0
ELF	    7fecabcad000-    7fecac053000	Export          libwine.so.1
ELF	    7fecac055000-    7fecac291000	Deferred        gameoverlayrenderer.so
ELF	    7fecac293000-    7fecac4bd000	Deferred        ld-linux-x86-64.so.2
Threads:
process  tid      prio (all id:s are in hex)
00000008 steam.exe
	00000024    0
	00000009    0
0000000c services.exe
	00000022    0
	0000001f    0
	0000001b    0
	00000018    0
	00000011    0
	0000000e    0
	0000000d    0
0000000f plugplay.exe
	00000015    0
	00000014    0
	00000010    0
00000016 winedevice.exe
	0000001c    0
	0000001a    0
	00000019    0
	00000017    0
0000001d winedevice.exe
	00000023    0
	00000021    0
	00000020    0
	0000001e    0
00000025 explorer.exe
	0000002b    0
	0000002a    0
	00000029    0
	00000026    0
0000002c (D) Z:\home\phusho\.local\share\Steam\steamapps\common\Path of Exile\PathOfExile_x64Steam.exe
	00000053    0
	00000050    1
	0000004f    1
	0000004e    0
	0000004d    0
	0000004c    0
	0000004b   15
	0000004a    2
	00000049    0
	00000048    0
	00000045    0 <==
	00000044    0
	00000043    0
	0000003a   15
	00000039    0
	00000038    0
	00000037    0
	00000033    0
	00000032    0
	00000031    0
	00000030    0
	0000002f    0
	0000002e    0
	0000002d    0
System information:
    Wine build: wine-4.11
    Platform: x86_64
    Version: Windows 7
    Host system: Linux
    Host version: 5.2.4-050204-generic
JS method call Messaging.PostMessage with 4 arguments
pid 18300 != 18299, skipping destruction (fork without exec?)
Game removed: AppID 238960 "", ProcID 18336 
Exiting app 238960
```

Interesting is this which trigger crash:
```
fExile_x64Steam.exe: ../src/amd/compiler/aco_insert_exec_mask.cpp:984: void aco::{anonymous}::process_block(aco::{anonymous}::exec_ctx&, aco::Block*): Assertion `block->index != ctx.program->blocks.size() - 1 || ctx.info[block->index].exec.size() <= 2' failed.
wine: Assertion failed at address 0x7fecab6dbe97 (thread 0045), starting debugger...
```

**Comment by aufkrawall** [comment]
It works on "real" Polaris here (RX 570).

**Comment by phush0** [comment]
so it is VegaM specific again

**Comment by Edgarins29** [comment]
It seems that I have same issue on RX 580 with ACO.
Works with previous Proton 4.2-9.

Strangely it worked OK in Blood Aquaducts, and in Act 4 town, but in Act 3 town it doesn't load.
Crash trigger entry is the same in my log file.

**Comment by daniel-schuermann** [comment]
does someone want to create a debug build, so that we know which assertion gets hit?
nvm, I should have scrolled to the right :P

**Comment by daniel-schuermann** [comment]
I cannot reproduce the bug, but the reason might be that I have only a lvl 5 character? :neutral_face:
You don't need to try to capture with llvm. it doesn't support shader_demote yet, thus uses different shaders...

**Comment by Edgarins29** [comment]
It was really strange for me.
At first I didn't have any problems, was farming one area all the time with one character with no issues.
At one point I refreshed this Github page and saw that there is issue and thought, hmm, I don't have any issues.
And then I relogged to another character and then after loading screen it crashed.

Afterwards I tried with high level and low level characters, each time the same assertion hits.

Also driver is the latest from PPA ACO version: 19.1.0+aco+git1564433637-5c6077221dd-1bionic2

Maybe there is some issue with that shader_demote at DXVK side or at least some inconsistency.

**Comment by daniel-schuermann** [comment]
Maybe your Proton updated mea while. I'll try again to reproduce this, tomorrow. 
I'm a bit confused how you can hit assertions with the ppa. It should be a release build... That said, does it work with release builds?

**Comment by Edgarins29** [comment]
First I updated to Proton 4.11 and then played.

I just tried rolling back to Mesa provided by Ubuntu 18.04. No issues.

Just reinstalled again ACO from PPA and then tried out Act 4 Highgate and Blood Aqueducts. Was working fine. Then moved to Act 2 Bridge Encampment and crashed.

Also Proton 4.2-9 is working fine and meanwhile can be played on that with ACO from PPA.

One thing I noticed that whenever move to old zones I played with older Proton crashes, but ones I played with the Proton 4.11 doesn't, but maybe that's coincidence. Have to check more.

**Comment by ernstp** [comment]
> One thing I noticed that whenever move to old zones I played with older Proton crashes, but ones I played with the Proton 4.11 doesn't, but maybe that's coincidence. Have to check more.

Shader cache invalidation issue?

**Comment by Edgarins29** [comment]
It was an idea, so I deleted all caches of shaders and the same areas crashes.

**Comment by Edgarins29** [comment]
i just tested running with Proton 4.11 and with PROTON_USE_WINED3D=1.
No crashes, but the performance is not that great.

Could this be some mismatch with this assertion and DXVK?

**Comment by daniel-schuermann** [comment]
No, that forces Proton to use OpenGL instead of Vulkan, so ACO isn't in use anymore. I'd rather use Proton 4.2 until we provide a patch.

**Comment by Edgarins29** [comment]
Ah, you're totally right!
Yeah, 4.2 for now.

Thank you!

**Comment by daniel-schuermann** [comment]
This problem should be fixed with 10690f8d9e659aa5caf0b9be7b48844ce0ac8dd3. If you encounter  further issues, feel free to open a new bug report. Thx!
